R/sacramento.R

```diff
@@ -7,110 +7,221 @@
+                    as.double(adimc_0),
+                    as.integer(min_ninc),
+                    NAOK = FALSE,
+                    DUP = FALSE,
+                    PACKAGE = "hydromad"
+                )
+                for (i in 7:25)
+                    attributes(states[[i]]) <- attributes(P)
+                ans <- do.call(cbind, states[7:25])
+                return(ans)
+            } else {
+                U <- .C(
+                    sma_sac,
+                    as.double(P),
+                    as.double(E),
+                    as.integer(NROW(DATA)),
+                    as.double(xpar),
+                    as.double(etmult),
+                    as.double(dt),
+                    U = double(NROW(DATA)),
+                    as.double(uztwc_0),
+                    as.double(uzfwc_0),
+                    as.double(lztwc_0),
+                    as.double(lzfsc_0),
+                    as.double(lzfpc_0),
+                    as.double(adimc_0),
+                    as.integer(min_ninc),
+                    NAOK = FALSE,
+                    DUP = FALSE,
+                    PACKAGE = "hydromad"
+                )$U
+                ## make it a time series object again
+                attributes(U) <- attributes(P)
+                ## re-insert missing values
+                U[bad] <- NA
+                return(U)
+            }
+        } else {
+            U <- rep(0, length(P))
+            sma = list(
+                uztwm = uztwm,
+                uzfwm = uzfwm,
+                uzk = uzk,
+                pctim = pctim,
+                adimp = adimp,
+                riva  = 0.0,
+                zperc = zperc,
+                rexp  = rexp,
+                lztwm = lztwm,
+                lzfsm = lzfsm,
+                lzfpm = lzfpm,
+                lzsk  = lzsk,
+                lzpk  = lzpk,
+                pfree = pfree,
+                rserv = 0.3,
+                side  = 0.0,
+                pxmlt = 1.0,
+                uztwc = uztwc_0 * uztwm,
+                uzfwc = uzfwc_0 * uzfwm,
+                lztwc = lztwc_0 * lztwm,
+                lzfsc = lzfsc_0 * lzfsm,
+                lzfpc = lzfpc_0 * lzfpm,
+                adimc = adimc_0 * (uztwm + lztwm),
+                min_ninc = min_ninc
+            )
+            
+            # SET SOME INITIAL VALUES TO ZERO
+            fsum1 = list(
+                srot = 0,
+                simpvt = 0,
+                srodt = 0,
+                srost = 0,
+                sintft = 0,
+                sgwfp = 0,
+                sgwfs = 0,
+                srecht = 0,
+                sett = 0,
+                se1 = 0,
+                se3 = 0,
+                se4 = 0,
+                se5 = 0
+            )
+            sma$epdist = etmult
+            sma$dt = dt
+            for (t in 1:NROW(DATA)) {
+                sma$ep =  E[t] * sma$epdist
+                sma$pxv = P[t] * sma$pxmlt
+                out <- fland1(sma,fsum1)
```

Comment:
```suggestion
                out <- fland1(sma,fsum1)
                fsum1 <- out$fsum1
```

Probably fsum1 needs to remember the output from the previous timestep too